### PR TITLE
chore(types): preserve historic transaction tags on chain upgrade

### PR DESCRIPTION
**Pull request by monitz87**
Also upgraded polkadot to latest version and upgraded types to latest mesh metadata

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,9 +84,9 @@
     "access": "public"
   },
   "dependencies": {
-    "@polkadot/api": "1.29.1",
-    "@polkadot/util": "3.2.1",
-    "@polkadot/util-crypto": "3.2.1",
+    "@polkadot/api": "2.3.1",
+    "@polkadot/util": "3.6.1",
+    "@polkadot/util-crypto": "3.6.1",
```
Comment (VictorVicente) [inline package.json:89]: Is there any breaking change from polkadot on these versions?

File: src/Polymesh.ts [hunk 2]
```diff
@@ -143,13 +144,20 @@ export class Polymesh {
 
     try {
       const { types, rpc } = polymesh;
+      console.log('HEY');
```
Comment (VictorVicente) [inline src/Polymesh.ts:147]: This should be removed.

File: src/Polymesh.ts [hunk 3]
```diff
@@ -143,13 +144,20 @@ export class Polymesh {
...
-        types,
+        // https://github.com/polkadot-js/api/releases/tag/v2.0.1 TODO @monitz87: remove once Polymesh is updated to substrate 2.0
+        types: {
+          ...types,
+          RefCount: 'RefCountTo259',
+        },
         rpc,
       });
 
+      console.log('HEY2');
```
Comment (VictorVicente) [inline src/Polymesh.ts:159]: This one too.

**Review by VictorVicente — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (monitz87) [reply package.json:89]: Already addressed

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**